quickwit/quickwit-doc-mapper/src/routing_expression/mod.rs

```diff
@@ -138,9 +160,16 @@ impl InnerRoutingExpr {
             InnerRoutingExpr::Composite(children) => {
                 ExprType::Composite.hash(hasher);
                 for child in children {
-                    child.eval_hash(ctx, hasher);
+                    child.eval_hash(ctx, hasher, salted_empty_hasher.clone());
                 }
             }
+            InnerRoutingExpr::Modulo(inner_expr, modulo) => {
+                ExprType::Modulo.hash(hasher);
+
+                let mut sub_hasher = salted_empty_hasher.clone();
```

Comment:
```suggestion
                let mut sub_hasher = H::default();
```

Reply:
Maybe we could get rid of the salt for the sub_hasher, just work off a `H::default()` and call it a day.